assets/js/extensions/jetpack/woocommerce-analytics/index.ts

```diff
@@ -0,0 +1,175 @@
+/**
+ * External dependencies
+ */
+import { Cart, isObject, objectHasProp } from '@woocommerce/types';
+import { select } from '@wordpress/data';
+import { getSetting } from '@woocommerce/settings';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY as CART_STORE_KEY } from '../../../data/cart/constants';
+
+declare global {
+	interface Window {
+		// eslint-disable-next-line @typescript-eslint/naming-convention
+		_wca: {
+			// eslint-disable-next-line @typescript-eslint/ban-types
+			push: ( properties: Record< string, unknown > ) => void;
+		};
+	}
+}
+
+/**
+ * Check if the _wca object is valid and has a push property that is a function.
+ *
+ * @param  wca {unknown} Object that might be a Jetpack WooCommerce Analytics object.
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+const isValidWCA = (
+	wca: unknown
+): wca is { push: ( properties: Record< string, unknown > ) => void } => {
+	if ( ! isObject( wca ) || ! objectHasProp( wca, 'push' ) ) {
+		return false;
+	}
+	return typeof wca.push === 'function';
+};
+
+const registerActions = (): void => {
```

Comment:
Not sure I understand this part.

Reply:
It is a placeholder for when we added shipping/payment method change events. If you like we can remove it from this PR or keep it so we have the code in place when it comes to adding them? Happy either way.